rllib/offline/estimators/weighted_importance_sampling.py

```diff
@@ -30,71 +31,88 @@ class WeightedImportanceSampling(OffPolicyEstimator):
     @override(OffPolicyEstimator)
     def __init__(self, policy: Policy, gamma: float):
         super().__init__(policy, gamma)
-        self.filter_values = []
-        self.filter_counts = []
+        # map from time to cummulative propensity values
+        self.cummulative_ips_values = []
+        # map from time to number of episodes that reached this time
+        self.episode_timestep_count = []
+        # map from eps id to mapping from time to propensity values
+        self.p = {}
+
+    @override(OffPolicyEstimator)
+    def estimate_on_single_episode(self, episode: SampleBatch) -> Dict[str, Any]:
+        estimates_per_epsiode = {}
+        rewards = episode["rewards"]
+
+        eps_id = episode[SampleBatch.EPS_ID][0]
+        if eps_id not in self.p:
+            raise ValueError(f"Episode {eps_id} not passed through the fit function")
```

Comment:
update the error message? it's not ``fit`` anymore? maybe say:
``"Can not find target weight for episode {eps_id}. Did it go though the peek_on_single_episode() function?"``?

Reply:
done